### ISSUE TITLE
Problem in the catch block of query.getObjects()

### ISSUE THREAD

**Issue by mfarizon** [knoxpo/dart_algolia#14]
Dart version: [2.4.0]
Flutter version: [1.7.8+hotfix.3 (stable)]
Algolia Dart Client version : [0.1.6+1]

**Describe the bug**
I think there is a problem with the error handling of the function getObjects(). It returns the catched error, which is creating an other exception :  'type 'SocketException' is not a subtype of type 'FutureOr\<AlgoliaQuerySnapshot\>', as it is trying to cast the error (SocketExcpetion here) into the return type (Future\<AlgoliaQuerySnapshot\>).  My guess is that it would be better to rethrow the error or handling the errors properly.

**To Reproduce**
Try calling getObjects on a query with no internet connection on the device.

**Expected behavior**
Throw typed Exceptions, in order to be able to catch the exception with 
```
try {
      AlgoliaQuerySnapshot snap = await query.getObjects();
} on SocketException catch(e) {
      print(e)
      // Error handling
}
```

**Comment by HiroyukTamura** [comment]
I've got same error.
looking forward for update.

**Comment by nhathiwala** [comment]
The update has been resolved in version 1.7, you may try updating.
